packages/ra-ui-materialui/src/layout/NotFound.tsx

```diff
@@ -45,8 +46,14 @@ function goBack() {
     window.history.go(-1);
 }
 
-const NotFound = props => {
-    const { className, classes: classesOverride, title, ...rest } = props;
+interface Props extends RouteComponentProps {
+    className: string;
+    title: TitleProps['defaultTitle'];
+    classes?: object;
+}
+
+const NotFound: FC<Props> = props => {
+    const { className, title, ...rest } = props;
```

Comment:
you should also add `classes: classesOverride` to the destructuring, or the root `<div>` may end up with a `classes` attribute which isn't DOM compliant.

Reply:
> > some component destructure classes property out I think we can remove this pattern and typescript will warn users if they pass a non-supported property
> 
> Not sure I understand. Can you give me an example?

that's what I was talking about, didn't know about `makeStyles` override behavior, Thank you